More column sizing adjustments

tmiw opened the pull request:
This PR performs the following to address feedback posted at https://github.com/drowe67/freedv-gui/pull/644#issuecomment-1879870187:

1. Limits displayed grid squares to six characters in length and actually ignores invalid input (vs. trying to calculate a distance that's probably wrong).
2. Instead of using wxWidgets' column autosizing, implement our own, ensuring that column sizes don't drop below hardcoded minimums.

tmiw commented:
@barjac, hopefully this works for you?

Tyrbiter commented:
I built and installed this latest version. There are some issues.

The 2 Mode columns are different widths for some reason. The SNR column is a bit too wide. The Locator column is now wider than the new maximum 6 characters displayed. The distance column could be narrowed a little, 5 digits should be sufficient.

I appreciate how difficult this is, it may also be affected by font sizes.

barjac commented:
Yes, here is a direct comparison of before this change and now:
![Screenshot_20240106_224900c](https://github.com/drowe67/freedv-gui/assets/1324702/e3849310-9a28-4e50-a0c1-6c3993cfbdc1)
![Screenshot_20240108_142420](https://github.com/drowe67/freedv-gui/assets/1324702/adfc553c-cd35-4a49-9e6f-7f3936f0adb2)

tmiw commented:
I fixed the Mode sizing discrepancy and added a change to the sizing calculation that might help. Unfortunately some space does need to be left for the sort indicator, so there might be a limit to how much further the other columns can shrink. For example, when sorting by SNR:

![Screenshot 2024-01-08 at 9 16 31 AM](https://github.com/drowe67/freedv-gui/assets/449242/b1f53ca7-e03c-4d0d-8954-09d87778ac4f)

Tyrbiter commented:
This seems much better, and apologies for forgetting about the space used by the column sorting indicator.

I don't think much improvement is possible now.

barjac commented:
I agree much improved :)
I have added another screen shot to the group above.

tmiw commented:
> While on the looks of the GUI, could a 'dark mode' be made available from within the program? I can use the system theme in plasma5 on x86_64 but changing the theme in LXQT DE on the RPi4 does not affect FreeDV. Maybe I just can't find the correct setting.

If you're using wxWidgets 3.2, it should automatically detect the OS/dev environment's setting for that. Seems to on Ubuntu with GNOME, anyway, albeit not 100% perfect. For example, when starting in dark mode:

![Screenshot 2024-01-08 at 11 07 43 PM](https://github.com/drowe67/freedv-gui/assets/449242/fa72ea56-b556-481d-b448-c6ce1c8aa590)

and then switching to light mode in the GNOME settings:

![Screenshot 2024-01-08 at 11 07 54 PM](https://github.com/drowe67/freedv-gui/assets/449242/d9d46e46-323d-4867-bbde-33a16a5d9755)

tmiw commented:
BTW, I just checked in a change that should make dark->light mode transitions work a bit better in the FreeDV Reporter window (at least on Linux and maybe macOS; Windows does have some separate logic for that control in wxWidgets so it might not have had the problem in the first place).
